packages/@icon-magic/icon-models/src/asset.ts

```diff
@@ -0,0 +1,87 @@
+import * as path from 'path';
+import * as debugGenerator from 'debug';
+import { AssetConfig, Content } from './interface';
+import { getFileContents } from './utils/files';
+
+/**
+ * This class abstracts the smallest set of information pertains to an asset
+ * file in memory
+ */
+export class Asset {
+  name: string;
+  contents: Content;
+  iconPath: string;
+  private debug: debugGenerator.IDebugger;
+  private unresolvedPath: string;
+
+  /**
+   * Creates a new asset
+   * @param iconPath Absolute path of the icon folder which this asset belongs
+   *                 to All paths in the asset are relative to this iconPath
+   * @param config a config object to set the initial properties of the asset
+   */
+  constructor(iconPath: string, config: AssetConfig) {
+    this.debug = debugGenerator('icon-magic:icon-models:asset');
+    // if iconPath is not absolute, throw an error
+    if (!path.isAbsolute(iconPath)) {
+      throw new Error(
+        'AssetCreationError: iconPath must always be absolute: ${iconPath)'
+      );
+    }
+    this.iconPath = iconPath;
+
+    // if the asset does not have a name, then set the name to be the name of
+    // the file itself
+    this.name = config.name ? config.name : path.parse(config.path).name;
+
+    // set the path (this automatically calls the setter defined on path)
+    this.path = config.path;
+
+    // set the contents only if it is passed in the config
+    if (config.contents) {
+      this.contents = config.contents;
+    }
+    this.debug(`Asset ${this.name} created in ${this.iconPath}`);
+  }
+
+  /**
+   * If an absolute path is passed in, store the relative path relative to the
+   * iconPath
+   */
+  set path(filePath: string) {
```

Comment:
Getters and setters are ✨magic ✨and don't make for the nicest APIs, especially when you can set a value, and get something completly different back. Consider using `getRelativePath()`, `getAbsolutePath()`, and `setPath(path: string)` for more explicit APIs.

Reply:
Can I do this as part of another fast follow?

Reply:
👍 Can you create a stub issue for it?